internal/osbuildexecutor/runner-impl-aws-ec2.go

```diff
@@ -24,6 +30,174 @@ func prepareSources(manifest []byte, store string, extraEnv []string, result boo
+
+	controlData := struct {
+		Exports []string `json:"exports"`
+	}{
+		Exports: exports,
+	}
+	controlDataBytes, err := json.Marshal(controlData)
+	if err != nil {
+		return "", err
+	}
+	err = os.WriteFile(control, controlDataBytes, 0600)
+	if err != nil {
+		return "", err
+	}
+	err = os.WriteFile(manifest, manifestData, 0600)
+	if err != nil {
+		return "", err
+	}
+
+	cmd := exec.Command("tar",
+		"-C",
+		cacheDir,
+		"-cf",
+		archive,
+		filepath.Base(control),
+		filepath.Base(manifest),
+	)
+	if output, err := cmd.CombinedOutput(); err != nil {
+		logrus.Errorf("Unable to create input tar: %v\n%s", err, output)
+		return "", err
+	}
+	/* #nosec G204 */
+	cmd = exec.Command("tar",
+		"-C",
+		filepath.Dir(store),
+		"-rf",
+		archive,
+		filepath.Base(store),
+	)
+	if output, err := cmd.CombinedOutput(); err != nil {
+		logrus.Errorf("Unable to create input tar: %v\n%s", err, output)
+		return "", err
+	}
+
+	return archive, nil
+}
+
+func handleBuild(inputArchive, host string) (*osbuild.Result, error) {
+	client := http.Client{
+		Timeout: time.Minute * 60,
+	}
+	inputFile, err := os.Open(inputArchive)
+	if err != nil {
+		return nil, err
+	}
+	defer inputFile.Close()
+
+	resp, err := client.Post(fmt.Sprintf("%s/api/v1/build", host), "application/x-tar", inputFile)
+	if err != nil {
+		logrus.Errorf("Unable to request build from executor instance: %v", err)
+		return nil, err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != 201 {
+		body, err := io.ReadAll(resp.Body)
+		if err != nil {
+			logrus.Warningf("Unable to read body waiting for build to run: %v", err)
+			return nil, err
+		}
+		return nil, fmt.Errorf("Something went wrong during executor build: %v\n%s", err, body)
+	}
+
+	var osbuildResult osbuild.Result
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("Something went wrong during executor build: %v\n%s", err, body)
+	}
+	err = json.Unmarshal(body, &osbuildResult)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to unmarshal response body into osbuild result: %v\nbody: %s", err, body)
+	}
+
+	return &osbuildResult, nil
+}
+
+func fetchOutputArchive(cacheDir, host string) (string, error) {
+	client := http.Client{
+		Timeout: time.Minute * 15,
+	}
+
+	resp, err := client.Get(fmt.Sprintf("%s/api/v1/result/output.tar", host))
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != 200 {
+		body, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return "", err
+		}
+		return "", fmt.Errorf("%v, body: %s", err, body)
+	}
+	file, err := os.Create(filepath.Join(cacheDir, "output.tar"))
+	if err != nil {
+		logrus.Errorf("Unable to write executor result tarball :%v", err)
+		return "", err
+	}
+	defer file.Close()
+	_, err = io.Copy(file, resp.Body)
+	if err != nil {
+		logrus.Errorf("Unable to write executor result tarball :%v", err)
+		return "", err
+	}
+	return file.Name(), nil
+}
+
+func extractOutputArchive(outputDirectory, outputTar string) error {
```

Comment:
Do we need to worry about about directory traversal attacks here? Given that things coming from the osbuild-worker-executor can be potentially tainted/toxic should we do a pre-check that there are no ".." or other unclean bits in the tar? Or is this already implicitly ensured? (*maybe* we could even extract as a different user but that would also be a followup).

Reply:
hmm, like first call "tar -tf" and see what's in it?

Reply:
Yeah, ideally we would be fine with a totally hostile remote side and my worry would be that they send us a tar full of ../../../../../../usr/bin/sshd with `#!/bin/sh -e p0wned`